### DISCUSSION TITLE
Rewriting execute commands for bazel (another) build system

### DISCUSSION THREAD

**Discussion by romannikov** [zed-industries/zed#15435] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

I'd like to develop an extension for zed for rewriting execute command (an arrow on the left side on the image) for bazel build system. 
I searched for some examples but honestly zed docs are not the best. Could you please give me some pointers how to do it?

<img width="667" alt="image" src="https://github.com/user-attachments/assets/732563f3-8cef-45d3-86cf-0c381ecf70e1">

**Comment by romannikov** [comment]
Basically I wanted to implement a lsp that will return `bazel build/run/test //..` commands. I'm not sure whether I'm right, but I did following steps:
1. Implemented test lsp
2. Assigned it for `.go` files
3. In Zed.log I can see that my lsp is launched. But also I see that gopls is not executed anymore. 

Are my actions correct?
Is it possible to launch two lsp? If not, how can I override Commands in the case of bazel build system?

**Comment by maherbeg** [comment]
Were you able to get this working at some point? I'm running into similar issues while trying to setup debugging for tests too

**Reply by romannikov** [reply]
No, I found that to support this I need to implement a bazel lsp or extend some of existing like Starlark. I started own lsp, but still have not enough time to finish it.

**Reply by ofek** [reply]
Are you able to share the progress you've made thus far?

**Reply by romannikov** [reply]
https://github.com/romannikov/bazel-lsp

It works in base cases, but I tested it only locally. I still doubt about the way of parsing BUILD files.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
